### PR TITLE
feat: api/metadata for ExpressionDimensionItem [DHIS2-14431]

### PR DESCRIPTION
**Pull request by d-bernat**
please see https://dhis2.atlassian.net/browse/DHIS2-14431

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/util/ExpressionDimensionItemUtils.java [hunk 1]
```diff
@@ -92,6 +92,45 @@ else if ( uids.length > 0 )
...
 
+    /**
+     * Provides expression validation
+     *
+     * @param manager {@link IdentifiableObjectManager} service for item
+     *        delivery
+     * @param expression or indicator of expression dimension item
+     * @return true when expression is valid
+     */
+    public static boolean validateExpressionItems( IdentifiableObjectManager manager, String expression )
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/util/ExpressionDimensionItemUtils.java:103]: if this method returns a boolean, I would find more appropriate to name like
`isValidExpressionItems`

**Review by gnespolino — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/util/ExpressionDimensionItemUtils.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/util/ExpressionDimensionItemUtils.java:103]: done

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/util/ExpressionDimensionItemUtils.java [hunk 2]
```diff
@@ -92,6 +92,45 @@ else if ( uids.length > 0 )
         return baseDimensionalItemObjects;
     }
 
+    /**
+     * Provides expression validation
+     *
+     * @param manager {@link IdentifiableObjectManager} service for item
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/util/ExpressionDimensionItemUtils.java:98]:
```suggestion
     * @param manager {@link IdentifiableObjectManager} service used to internally load required objects
```

**Review by maikelarabori — COMMENTED** [review]
Thanks, @d-bernat. I left a few comments for review.

Cheers,
Maikel

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/util/ExpressionDimensionItemUtils.java [hunk 3]
```diff
@@ -92,6 +92,45 @@ else if ( uids.length > 0 )
         return baseDimensionalItemObjects;
     }
 
+    /**
+     * Provides expression validation
```
[the remark this answers is not in the document]
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/util/ExpressionDimensionItemUtils.java:96]: done

File: dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/util/ExpressionDimensionItemUtilsTests.java [hunk 4]
```diff
@@ -25,34 +25,46 @@
...
 import org.junit.jupiter.params.provider.CsvSource;
 import org.mockito.Mock;
 
 /**
- * Test for {@link ExpressionDimensionItemHelper}
+ * Test for {@link ExpressionDimensionItemUtils}
  */
-class ExpressionDimensionItemHelperTests
+@ExtendWith( org.mockito.junit.jupiter.MockitoExtension.class )
+class ExpressionDimensionItemUtilsTests
```
[the remark this answers is not in the document]
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/util/ExpressionDimensionItemUtilsTests.java:53]: change to service, done

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/expressiondimensionitem/ExpressionDimensionItemService.java [hunk 5]
```diff
@@ -25,44 +25,45 @@
...
 import org.hisp.dhis.dataelement.DataElementOperand;
+import org.springframework.stereotype.Service;
 
 /**
  * Parsing the expression of ExpressionDimensionItem, provides collection of
  * BaseDimensionalItemObjects.
  */
-public class ExpressionDimensionItemHelper
+@Service
+@AllArgsConstructor
```
Comment (vietnguyen) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/expressiondimensionitem/ExpressionDimensionItemService.java:50]: We avoid using `AllArgsConstructor`, and try to use `RequiredArgsConstructor` if possible.

**Review by vietnguyen — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/expressiondimensionitem/ExpressionDimensionItemService.java (hunk 5 above)
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/expressiondimensionitem/ExpressionDimensionItemService.java]: done

**Review by maikelarabori — APPROVED** [review]

**Pull request merged**